### ISSUE TITLE
Extend adds 0s instead of 255s for INT1U rasters

### ISSUE THREAD

**Issue by JimShady** [rspatial/terra#620]
I am looking to use the terra::extend function. My raster is datatype INT1U (where nodata is specified by the value 255), when I use extend the new empty cells are given the value 0. Which is problematic for my workflow (as I have other cells which have the value 0 but are not NA).

```
r <- rast(xmin=-150, xmax=-120, ymin=30, ymax=60, ncols=3, nrows=3)
values(r) <- c(0,1,2,3,4,5,6,7,255)

r <- writeRaster(r, filename = "test.tif", datatype = "INT1U", overwrite=T)

e <- ext(-180, -100, 40, 70)

re <- extend(r, e, file = "test2.tif", datatype = "INT1U", overwrite=T)

plot(re)
```

![image](https://user-images.githubusercontent.com/2901470/165723424-d1820c76-cab3-475d-bf9b-0fdc87098f57.png)

Interestingly @rhijmans noticed in our [StackExchange discussion](https://stackoverflow.com/questions/72034716/extend-from-terra-fills-with-0-instead-of-255/72038533#72038533) that if you extend in memory, and then write the file out, the behavior is different.

```
r <- rast(xmin=-150, xmax=-120, ymin=30, ymax=60, ncols=3, nrows=3)
values(r) <- c(0,1,2,3,4,5,6,7,255)
r <- writeRaster(r, filename = "test.tif", datatype = "INT1U", overwrite=T)
e <- ext(-180, -100, 40, 70)

re <- extend(r, e)
re <- writeRaster(re, file = "test2.tif", datatype = "INT1U", overwrite=T)
plot(re)
```

So it seems that for INT1U datatypes, extend is not using 255 for adding empty cells when writing to file?

![image](https://user-images.githubusercontent.com/2901470/165724869-2cda898e-ca53-45d1-b68c-120da8f2fa20.png)

**Comment by rhijmans** [comment]
Thank you. I believe has now been fixed.

**Comment by JimShady** [comment]
Great, thanks @rhijmans . I'd like to start using this new version as soon as possible. If I use the below command,  should I get the fixes? I think so?

`remotes::install_github("rspatial/terra")`

**Comment by kadyb** [comment]
Yes. You can also download a development binary version (no compilation required) for Windows and MacOS from R-universe:

```r
install.packages("terra", repos = "https://rspatial.r-universe.dev")
```
